core/src/main/scala/kafka/tools/ConsoleConsumer.scala

```diff
@@ -456,8 +460,20 @@ object ConsoleConsumer extends Logging {
       KafkaMetricsReporter.startReporters(verifiableProps)
     }
 
-    //Provide the consumer with a randomly assigned group id
-    if (!consumerProps.containsKey(ConsumerConfig.GROUP_ID_CONFIG)) {
+    // group id must not be provided more than once
+    val groupIdDirectPresent = options.has(groupIdOpt)
+    val groupIdPresentCount = Seq(
```

Comment:
I think it's fine if the group is provided in multiple places as long as it's the same value. What do you think?

Reply:
It seems that similar tools do not set restrictions when it comes to repeating property values through different means. For example, I just tried the console producer, and provided a custom `batch.size` through `--producer-property`, along with a different direct value through `--max-partition-memory-bytes`. The tool does not complain about this, and simply gives priority to the directly provided config. Sorry for trying this late, but if we want to remain consistent the original commit would be more reasonable.

Reply:
Hmm... I think those cases are a bit different. If we use an unexpected batch size, no harm done really. But if we use the wrong consumer group, it could cause some damage. I think it's reasonable to throw an error back at the user if there's any inconsistency with passed group ids.

Reply:
That makes sense. I'll submit another patch to allow providing the same group id in different ways.